/**
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package protoPanda;

message DebuginfoIns {
    uint64 lineNumber = 1;
    uint32 columnNumber = 2;
    bytes wholeLine = 3;
    uint64 boundLeft = 4;
    uint64 boundRight = 5;
}

message LocalVariable {
    bytes name = 1;
    bytes signature = 2;
    bytes signatureType = 3;
    int32 reg = 4;
    uint32 start = 5;
    uint32 length = 6;
}
